resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientProxy.java

```diff
@@ -52,6 +56,34 @@ public Object invoke(Object o, Method method, Object[] args)
                 return this.toString();
             } else if (method.getName().equals("as") && args.length == 1 && args[0] instanceof Class) {
                 return ProxyBuilder.proxy((Class<?>) args[0], target, config);
+            } else if (method.isDefault()) {
+                // We need to use the privateLookupIn and lookup(), compared to publicLookup(), as the proxy is likely
+                // in a different module. See the JavaDoc for MethodHandles.privateLookupIn() for details on how this
+                // works and this allows more permissive rules.
+                if (System.getSecurityManager() == null) {
+                    return MethodHandles.privateLookupIn(clazz, MethodHandles.lookup())
+                            .unreflectSpecial(method, clazz)
+                            .bindTo(o)
+                            .invokeWithArguments(args);
+                }
+                try {
+                    return AccessController.doPrivileged((PrivilegedExceptionAction<Object>) () -> {
+                        try {
+                            return MethodHandles.privateLookupIn(clazz, MethodHandles.lookup())
+                                    .unreflectSpecial(method, clazz)
+                                    .bindTo(o)
+                                    .invokeWithArguments(args);
+                        } catch (Throwable e) {
```

Comment:
It looks like this could catch something like `OutOfMemoryError`, `StackOverflowError` or `VirtualMachineError` and wrap it in `RuntimeException`. Should this only wrap Exception types and rethrow other error types as-is?

Reply:
It would seemingly also wrap unchecked exception types like `jakarta.ws.rs.NotFoundException` which callers might expect to observe directly.

Reply:
Yes. It's likely better to only wrap `Exception` types and re-throw any `WebApplicationException`'s. Nice, catch, thank you!